src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -139,17 +139,73 @@ struct sdm_sensor_info
   using data_type = query::sdm_sensor_info::data_type;
   using sdr_req_type = query::sdm_sensor_info::sdr_req_type;
 
+  static data_type
+  parse_sysfs_nodes(const xrt_core::device* device,
+                    std::string tpath,
+                    bool *next_id)
+  {
+    auto pdev = get_pcidev(device);
+    //sensors are stored in hwmon sysfs dir with name ends with as follows.
+    std::array<std::string, 6> sname_end = {"label", "input", "max", "average", "highest", "status"};
+    int max_end_types = sname_end.size();
+    data_type data = { 0 };
+
+    for (int end_id = 0; end_id < max_end_types; end_id++)
+    {
+      std::string target_snode = tpath + sname_end[end_id];
+      if (end_id == 0)
```

Comment:
// label case I guess?
The numbers really obfuscate and prevent this code from being readable.  Comments would help.